app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -874,7 +874,9 @@ class BrowserTabFragment :
             object : DefaultLifecycleObserver {
                 override fun onStop(owner: LifecycleOwner) {
                     if (isVisible) {
-                        updateOrDeleteWebViewPreview()
+                        if (viewModel.browserViewState.value?.maliciousSiteDetected != true){
+                            updateOrDeleteWebViewPreview()
```

Comment:
We need to delete the webview preview as well. Otherwise, there's some scenarios where the previous preview will still be shown. I managed to reproduce it by loading wikipedia -> phishing page with no forced delay. I'd just extract the logic to delete the preview from updateOrDeleteWebViewPreview, and call if from the else branch.